packages/block-editor/src/components/block-list/block.js

```diff
@@ -544,13 +559,38 @@ function BlockListBlock( {
 						ref={ breadcrumb }
 					/>
 				) }
-				{ ( shouldShowContextualToolbar || isForcingContextualToolbar.current ) && (
+
+				{ ( isCapturingDescendantToolbars || captureChildToolbar ) && (
+					// A slot made available on all ancestors of the selected Block
+					// to allow child Blocks to render their toolbars into the DOM
+					// of the appropriate parent.
+					<ChildToolbarSlot />
+				) }
+
+				{ ( ( ! ( captureChildToolbar || hasAncestorCapturingToolbars ) && shouldShowContextualToolbar ) || isForcingContextualToolbar.current ) && (
+					// Standard toolbar attached directly to the Block.
 					<BlockContextualToolbar
 						// If the toolbar is being shown because of being forced
 						// it should focus the toolbar right after the mount.
 						focusOnMount={ isForcingContextualToolbar.current }
 					/>
+
 				) }
+
+				{ ( captureChildToolbar || hasAncestorCapturingToolbars ) && shouldShowContextualToolbar && (
+					// If the parent Block is set to consume toolbars of the child Blocks
+					// then render the child Block's toolbar into the Slot provided
+					// by the parent.
+					<ChildToolbar>
+						<BlockContextualToolbar
```

Comment:
Should we render BlockContextualToolbar in a variable when it is going to be used to avoid duplicating the code?

Reply:
I've updated this.